libgnucash/app-utils/gnc-helpers.c

```diff
@@ -133,3 +134,35 @@ gnc_quoteinfo2scm(gnc_commodity *comm)
     info_scm = scm_cons (name ? scm_from_utf8_string (name) : SCM_BOOL_F, info_scm);
     return info_scm;
 }
+
+Account * gnc_account_get_linked_account (const Account *acc, const gchar *id)
+{
+    GValue v = G_VALUE_INIT;
+    GncGUID *guid = NULL;
+
+    g_return_val_if_fail (GNC_IS_ACCOUNT (acc), FALSE);
+
+    qof_instance_get_kvp (QOF_INSTANCE(acc), &v, 1, id);
```

Comment:
`qof_instance_get` calls `g_object_get_valist` which calls `g_object_get_is_valid_property` and throws a warning for an account without these slots. I cannot find a suitable API to test that g_object has property. Any ideas?

```
sys:1: Warning: g_object_get_is_valid_property: object class 'Account' has no property named 'proceeds_account'
sys:1: Warning: g_object_get_is_valid_property: object class 'Account' has no property named 'dividend_account'
sys:1: Warning: g_object_get_is_valid_property: object class 'Account' has no property named 'capgains_account'
sys:1: Warning: g_object_get_is_valid_property: object class 'Account' has no property named 'fees_account'
```

Reply:
fwiw the `qof_object_get/set` (as written in current branch) doesn't seem to work.

Reply:
> Any ideas?
Yes. Declare the properties by adding them to the [properties enum](https://github.com/Gnucash/gnucash/blob/master/libgnucash/engine/Account.cpp#L85), call `g_object_class_install_property` on each of them in [`gnc_account_class_init`](https://github.com/Gnucash/gnucash/blob/master/libgnucash/engine/Account.cpp#L645), and add getters and setters to [`gnc_account_get_property`](https://github.com/Gnucash/gnucash/blob/master/libgnucash/engine/Account.cpp#L368) and [`gnc_account_set_property`](https://github.com/Gnucash/gnucash/blob/master/libgnucash/engine/Account.cpp#L516) respectively. You don't need to write explicit functions, see the examples of using `qof_instance_get_path` and `qof_instance_set_path` near the bottom of `gnc_account_get/set_property`.

Reply:
> fwiw the qof_object_get/set (as written in current branch) doesn't seem to work.

No surprise, there's no such functions. It's `qof_instance_get/set`, and considering how much they're used it would be a major problem if they don't work:
```
grep -rl \\bqof_instance_get\\b *gnucash
gnucash/gnome/dialog-sx-editor.c
gnucash/gnome/gnc-split-reg.c
gnucash/gnome/dialog-sx-editor2.c
gnucash/gnome/gnc-plugin-page-register2.c
gnucash/gnome/dialog-invoice.c
gnucash/gnome/dialog-payment.c
gnucash/gnome/gnc-plugin-page-report.c
gnucash/register/ledger-core/split-register-model.c
gnucash/import-export/test/gtest-import-backend.cpp
gnucash/import-export/ofx/gnc-ofx-import.c
gnucash/import-export/import-utilities.c
gnucash/import-export/aqb/gnc-ab-kvp.c
gnucash/gnome-utils/dialog-preferences.c
gnucash/gnome-utils/gnc-tree-util-split-reg.c
gnucash/gnome-utils/gnc-tree-view-account.c
libgnucash/engine/mocks/gmock-qofinstance.cpp
libgnucash/engine/gncInvoice.c
libgnucash/engine/gncOwner.c
libgnucash/engine/test/test-engine-kvp-properties.c
libgnucash/engine/gnc-budget.c
libgnucash/engine/gnc-lot.c
libgnucash/engine/cap-gains.c
libgnucash/engine/qofinstance.cpp
libgnucash/engine/SX-book.c
libgnucash/engine/qofinstance.h
libgnucash/engine/qofbook.cpp
libgnucash/app-utils/gnc-accounting-period.c
libgnucash/app-utils/gnc-sx-instance-model.c
```